common/ctree.h

```diff
@@ -323,6 +354,37 @@ class ctree
 			}
 		}
 
+		void merge(const node_base_t* other) override
+		{
+			for (const auto& [key, node] : reinterpret_cast<const node_t*>(other)->next)
+			{
+				if (auto it = next.find(key); it != next.end())
+				{
+					it->second->merge(node);
+				}
+				else
+				{
+					if constexpr (sizeof...(next_counters_T) != 0)
+					{
+						using actual = node_t<next_counters_T...>;
+						next.emplace(key, new actual(*reinterpret_cast<actual*>(node)));
+					}
+					else
+					{
+						next.emplace(key, new node_base_t(*node));
+					}
+				}
+			}
+
+			for (const auto& [key, value] : reinterpret_cast<const node_t*>(other)->convert)
+			{
+				if (convert.find(key) == convert.end())
+				{
+					convert.emplace(key, value);
+				}
+			}
+		}
```

Comment:
I think we can just use `static_cast` like in the rest of the code. I don't like `reinterpret cast`.
Smth like:
```cpp
const auto* other_node = static_cast<const node_t*>(other);
//...
for (const auto& [key, node] : other_node->next)
{
    //..
		next.emplace(key, new actual(*static_cast<actual*>(node)));
//..
for (const auto& [key, value] : other_node->convert)
//...
```